tests/src/rules/extensions.js

```diff
@@ -613,6 +613,14 @@ describe('TypeScript', () => {
             ],
             parser,
           }),
+          test({
+            code: 'export type T from "./typescript-declare";',
```

Comment:
Hi guys, I think the code here should be named export instead in order to be valid. Test failed because parser couldn't understand what exactly it is trying to export.

```suggestion
            code: 'export type { T } from "./typescript-declare";',
```

Reply:
@ljharb @Lukas-Kullmann Could you guys take a look at this ? I think these changes should make this PR ready.